### PR TITLE
[#13] [Android] [Integrate] As a user, I can see the Home screen header

### LINKED ISSUE #13: [Android] [Integrate] As a user, I can see the Home screen header
## Why

After logging in, the user can see the Home screen with the user profile info.

## Acceptance Criteria

- Show the date-time in the left corner.
- Show the avatar on the right.

### PR DESCRIPTION
**Pull request by luongvo**
- Close #13

## What happened 👀

After logging in, the user can see the Home screen with the user profile info:
- Show the `current date` in the left corner.
- Show the `user avatar` on the right.
- ~Refactor: body1 -> subtitle1, body2 -> subtitle2, subtitle1 -> body1.~ 👉  DROPPED
- Show an error message in case loading user project fails.
- Add UI tests for Home screen's header.

## Insight 📝

- As recommended in the latest update of `lifecycle-runtime-compose`, we should consume flows safely with [`collectAsStateWithLifecycle`](https://developer.android.com/reference/kotlin/androidx/lifecycle/compose/package-summary#(kotlinx.coroutines.flow.Flow).collectAsStateWithLifecycle(kotlin.Any,androidx.lifecycle.Lifecycle,androidx.lifecycle.Lifecycle.State,kotlin.coroutines.CoroutineContext)) https://medium.com/androiddevelopers/consuming-flows-safely-in-jetpack-compose-cde014d0d5a3.
- I added `DateFormatter` helper to make testing in HomeViewModel easier by mocking the current date data.
- Show error message as `Snackbar` https://medium.com/@jurajkunier/how-to-show-snackbar-in-jetpack-compose-3f2d81891f87.

## Proof Of Work 📹

- Success

- Failed to load user profile (no internet)

https://user-images.githubusercontent.com/16315358/209897324-7020a29b-45aa-4d2e-b422-b48017b198c8.mp4

### CODE REVIEW

**Review by blyscuit — APPROVED** [review]

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt [hunk 1]
```diff
@@ -1,23 +1,52 @@
...
+    val avatarUrl by viewModel.avatarUrl.collectAsStateWithLifecycle()
+
+    val scaffoldState: ScaffoldState = rememberScaffoldState()
+    val context = LocalContext.current
+    LaunchedEffect(viewModel.error) {
+        viewModel.error?.let {
+            scaffoldState.snackbarHostState.showSnackbar(
+                message = it.userReadableMessage(context)
+            )
+            viewModel.error = null
```
Comment (manh-t) [inline android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt]: we should not modify the viewmodel's variable directly in Compose 🤔 I understand why you do that with this approach, but we should avoid using it IMO. How about [this approach](https://stackoverflow.com/a/69354712)? 🙆‍♂️ Feel free to discuss to find a good solution for this 🙇

**Review by doannimble — APPROVED** [review]
LGTM 🏃‍♂️

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt (hunk 1 above)
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt]: @manh-t I replied you on Slack instead https://nimble-co.slack.com/archives/C04BVRP6UDR/p1672386758164719 👋

**Comment by luongvo** [comment]
FYI: I dropped this change `Refactor: body1 -> subtitle1, body2 -> subtitle2, subtitle1 -> body1.` 👉 DROPPED

File: android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt (hunk 1 above)
Comment (luongvo) [reply android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt]: @manh-t Fixed in https://github.com/luongvo/kmm-survey/pull/62/commits/74ac75a08423c5ff5a73d5fc68402b6e316de020

**Review by manh-t — APPROVED** [review]
🚀

**Pull request merged**